src/lib/stores/player.ts

```diff
@@ -69,6 +86,21 @@ export function createPlayerStore() {
 		set(createInit());
 	};
 
+	const handlePaused = (paused: boolean) => {
+		if (paused) {
+			clearInterval(progressTimer);
+		} else {
+			progressTimer = createInterval();
+		}
+	};
+
+	const createInterval = (): NodeJS.Timer =>
+		setInterval(() => {
+			update((currentState) => ({ ...currentState, progress: currentState.progress + 1 }));
```

Comment:
regarding the other comment. this line would be just guarded by an if checking if the song is paused